lib/basevm.py

```diff
@@ -15,29 +15,13 @@ def __init__(self, uuid, host):
         self.host = host
 
     def param_get(self, param_name, key=None, accept_unknown_key=False):
-        return _param_get(self.host, self.xe_prefix, self.uuid, param_name, key, accept_unknown_key)
+        return _param_get(self.host, BaseVM.xe_prefix, self.uuid, param_name, key, accept_unknown_key)
```

Comment:
I think using `type(self)` instead of `BaseVM` would make the code more generic and flexible. However I'm not sure if there are drawbacks to this approach.

Reply:
AFAIK this is common practice + easier to read IMHO.